### PR TITLE
Updated Intl FormattedMessage component mock.

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Related to #2786

## Related Issue

Closes #ISSUE_NUMBER.

### Verification Steps

1. Go to the FOO page.
2. Verify the BAR shows up.
3. Make sure BAZ does a thing.

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/__mocks__/react-intl.js [hunk 1]
```diff
@@ -7,6 +8,8 @@ const intl = reactIntl.createIntl({
 
 module.exports = {
     ...reactIntl,
-    FormattedMessage: jest.fn(({ defaultMessage, id }) => defaultMessage || id),
+    FormattedMessage: props => (
+        <div componentName="Formatted Message Component" {...props} />
```
Comment (tjwiebell) [inline packages/venia-ui/__mocks__/react-intl.js:12]: Jest suggests mock components using this convention, which I think I'd prefer so we're not adding semantic markup.

https://jestjs.io/docs/en/tutorial-react#snapshot-testing-with-mocks-enzyme-and-react-16

```suggestion
        <mock-FormattedMessage>" {...props} />
```

File: packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap [hunk 2]
```diff
@@ -48,7 +48,7 @@ Array [
     1
   </span>,
   <span>
-     
+
```
Comment (tjwiebell) [inline packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap:51]: Why do I feel like I've seen this several times now; any idea why the whitespace in this snap keeps changing?

Comment (revanth0212) [reply packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap:51]: It keeps sneaking in every time I update the mocks. Thanks for catching it, Tommy. I'll have to research and fix this once and for all. Even @jcalcaben has this issue.

**Review by tjwiebell — CHANGES_REQUESTED** [review]
I support this change, let's figure out how to get into the queue; get rid of the `div` usage and this looks ready.

**Review by tjwiebell — APPROVED** [review]
Love it 👍  let's chat about it during team time and get this merged.

File: packages/venia-ui/__mocks__/react-intl.js [hunk 3]
```diff
@@ -7,6 +8,6 @@ const intl = reactIntl.createIntl({
 
 module.exports = {
     ...reactIntl,
-    FormattedMessage: jest.fn(({ defaultMessage, id }) => defaultMessage || id),
+    FormattedMessage: props => <mock-FormattedMessage {...props} />,
```
Comment (sirugh) [inline packages/venia-ui/__mocks__/react-intl.js:11]: I really like this pattern for mocking things! Going to start using it.

**Pull request merged**